src/pkg/clouds/aws/ecs/logs.go

```diff
@@ -29,39 +28,40 @@ func getLogGroupIdentifier(arnOrId string) string {
 func QueryAndTailLogGroups(ctx context.Context, start, end time.Time, logGroups ...LogGroupInput) (LiveTailStream, error) {
 	ctx, cancel := context.WithCancel(ctx)
 
-	e := &eventStream{
-		cancel: cancel,
-		ch:     make(chan types.StartLiveTailResponseStream),
-	}
+	errCh := make(chan error)
 
-	// We must close the channel when all log groups are done
-	var wg sync.WaitGroup
-	var err error
+	var eventCh chan LogEvent
 	for _, lgi := range logGroups {
-		var es LiveTailStream
-		es, err = QueryAndTailLogGroup(ctx, lgi, start, end)
+		es, err := QueryAndTailLogGroup(ctx, lgi, start, end)
 		if err != nil {
-			break // abort if there is any fatal error
+			cancel()
+			return nil, err
```

Comment:
We're not closing `eventCh` here. Do we care?

Reply:
We should not close a channel that we are reading from, it is the sender's responsibility. In this case it is done by the Merge function. And the context cancellation will stop all the senders and triggers the close.